File: Python/ByRound/2057/2057_C_Trip_to_the_Olympiad.py
# Problem: CF 2057 C - Trip to the Olympiad
# https://codeforces.com/contest/2057/problem/C

"""
Purpose: To find three distinct integers a, b, c in range [l, r] that maximize the expression (a⊕b)+(b⊕c)+(a⊕c),
where ⊕ is the XOR operation.

Algorithm:
- The key idea is to maximize XOR values between pairs of numbers.
- We determine the highest bit position where l and r differ.
- Using this bit position, we construct three numbers such that their pairwise XORs are maximized.
- The construction involves:
    1. Finding a number `a` that is just below or equal to `r` but with the highest possible differing bits set.
    2. Setting `b` as `a + 1`.
    3. Setting `c` as either `l` or `r`, depending on whether `a` or `b` equals `r`.

Time Complexity: O(1) - Constant time due to fixed bit-length operations.
Space Complexity: O(1) - Only using a constant amount of extra space.
"""

from sys import stdin, stdout

input = stdin.readline

for _ in range(int(input())):
    l, r = map(int, input().split())
    # Determine the bit position where l and r first differ, starting from the highest bit.
    k = (l ^ r).bit_length() - 1
    # Construct 'a' by setting all bits up to k to 1, starting from l
    a = l | ((1 << k) - 1)
    # Increment a to get b (this ensures b differs from a at the k-th bit)
    b = a + 1
    # Choose c such that it's either l or r, avoiding conflict if a or b is r
    c = l if a == r or b == r else r
    stdout.write(f"{a } {b } {c } \n")


# https://github.com/VaHiX/CodeForces/

File: CPP/ByTier/C/2057_C_Trip_to_the_Olympiad.cpp
// Problem: CF 2057 C - Trip to the Olympiad
// https://codeforces.com/contest/2057/problem/C

/*
 * Problem: C. Trip to the Olympiad
 * 
 * Purpose: To find three distinct integers a, b, c within range [l, r] 
 * such that the expression (a⊕b)+(b⊕c)+(a⊕c) is maximized.
 * 
 * Algorithm:
 * - The key insight is to maximize the XOR sum by choosing numbers 
 *   that differ in as many bits as possible at each position.
 * - We compute the highest bit where l and r differ, k.
 * - Then construct a, b, c using this bit to form combinations that 
 *   yield maximum XOR values.
 * 
 * Time Complexity: O(1) per test case
 * Space Complexity: O(1)
 */

#include <stdio.h>
#include <iosfwd>

using namespace std;
typedef long long ll;
int T, l, r;
int main() {
  scanf("%d", &T);
  while (T--) {
    scanf("%d%d", &l, &r);
    int k = 31 - __builtin_clz(l ^ r);  // Find highest differing bit position between l and r
    int a = l | (1 << k) - 1;           // Construct a such that it has 1s from bit k onwards
    printf("%d %d %d\n", a, a + 1, a == l ? r : l);  // Output a, a+1, and either l or r depending on a
  }
  return 0;
}


// https://github.com/VaHiX/CodeForces/